File: Hosts/TravelRepublic.Angular/src/app/hotels/requests/hotel-search-filter-request.ts
import { HotelSearchRequestBase } from './hotel-search-request-base';
export class HotelSearchFilterRequest extends HotelSearchRequestBase {
    constructor(public Name = '',
                public Star = 0,
                public UserRating = 0,
                public CostMin = 0,
                public CostMax = 0) {
        super(Name, Star, UserRating, CostMin, CostMax);
    }
}

File: Business/TravelRepublic.Business/Requests/HotelSearchFilterRequest.cs
using Eml.Mediator.Contracts;
using TravelRepublic.Business.Responses;

namespace TravelRepublic.Business.Requests
{
    public class HotelSearchFilterRequest : HotelSearchRequestBase, IRequestAsync<HotelSearchFilterRequest, HotelSearchFilterResponse>
    {
        public HotelSearchFilterRequest(string name, int star, double userRating, double costMin, double costMax)
            : base(name, star, userRating, costMin, costMax)
        {
        }
    }
}
